### ISSUE TITLE
[Bug]: WCO buttons have additional spacing in v20

### ISSUE THREAD

**Issue by rzhao271** [electron/electron#35312]
Alright, I guess I wasn't running it in the right electron version, but with fiddle it also is off-center. This issue doesn't appear in v20.0.0, but in v20, the WCO just looks weird. Maybe that bug occurs only with some specific settings, I'll look into it later.

This is what I mean by "weird":
![screenshot](https://user-images.githubusercontent.com/90904039/182640528-d0d5e3fa-3180-4254-8cd7-4e67a699ea64.png)
![screenshot](https://user-images.githubusercontent.com/90904039/182640532-2054adeb-9596-4990-8c04-c1ff15535ee7.png)

_Originally posted by @wheezard in https://github.com/electron/electron/issues/35038#issuecomment-1204061897_

**Comment by codebytere** [comment]
@rzhao271 yeah it's https://github.com/electron/electron/commit/2c0327c3948a157a0ab15248322bc2941407465a - the `GetBetweenButtonSpacing()` change. I can look into it, i don't know why it looks different than Chrome with that same code.

**Comment by rzhao271** [comment]
I think it could have to do with the background colour.
The colour Chromium is using: https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/ui/views/frame/windows_caption_button.cc;l=80

The colour being set to transparent: https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/ui/color/chrome_color_mixer.cc;l=211

What Electron is doing:
https://github.com/electron/electron/pull/34790/files#diff-2a87d5d8844989448e4571deacfd3f396b57cb5d9cdcac1d2b725e96ca57261fR60

**Comment by bilalshaikh292** [comment]
hello, I would like to work on this issue

**Comment by samuelmaddock** [comment]
Repro gist: https://gist.github.com/73c0096a360af66168e93ffe5d508c04

| 20.0.0-beta.8 | 20.0.0-beta.9 |
| --- | --- |
| ![image](https://user-images.githubusercontent.com/1656324/184971449-b3ccdb1f-e6f8-4f75-a260-e40b665d9914.png) | ![image](https://user-images.githubusercontent.com/1656324/184971586-01307bef-dff3-48a0-b035-9f1557e3a11f.png) |

**Comment by clavin** [comment]
It looks like the raw window render is showing through the gaps, meaning the buttons probably aren't being sized properly. I suspect it was this one line in this commit: https://github.com/electron/electron/commit/2c0327c3948a157a0ab15248322bc2941407465a#diff-2a87d5d8844989448e4571deacfd3f396b57cb5d9cdcac1d2b725e96ca57261fR64

**Comment by jediwade** [comment]
Wasn't sure if this would help or not, but what I remember noticing is that WCO has issues (like @samuelmaddock included screenshots of) when the monitor that the app is on is not at 100% DPI. I can drag my app window between two displays (1 at 100% DPI, the other at 150% DPI) and it will go back and forth between having rendering issues and not having rendering issues. This is what I remember a week ago doing testing on another bug, but I can re-confirm tonight if wanted. My computer is running latest non-preview build of Windows 11 if that matters.

**Comment by rzhao271** [comment]
It really is that whatever is behind the buttons is showing through. I loaded up a webpage to test something else and noticed the webpage was showing behind the buttons. This matches with what @samuelmaddock and @clavin noticed.
But, on top of that, I get rendering glitches by maximizing and restoring the window, so the gaps aren't even being rendered consistently.

![Screenshot showing WCO with transparent background between buttons](https://user-images.githubusercontent.com/7199958/185664126-fe5726f8-5d5d-4181-9c1a-caeac6b99617.PNG)
![Screenshot showing WCO with rendered background between buttons](https://user-images.githubusercontent.com/7199958/185664128-ce9f4058-5e28-43ac-befb-835224a15fb5.PNG)

A question I have is what should Electron do for the gaps?
Should it use the background colour passed in to the BrowserWindow ctor, try to maintain transparency between the buttons, or remove the spacing altogether?

Note that Chromium actually has gaps between the buttons https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/ui/views/frame/windows_caption_button.cc;drc=6c0349ec4fde5dadd483f2ed36f73e494595b934;l=144?q=windows_caption_button.&ss=chromium.

**Comment by indutny-signal** [comment]
This is how it looks like for us on Windows 11
<img width="738" alt="image" src="https://user-images.githubusercontent.com/79877362/186783053-05831118-db9c-4e6c-a398-683bb9d21518.png">

At least in our case it doesn't look like the pixels in-between the buttons are from the background...

**Comment by arman-canva** [comment]
+1

Any idea whether this is likely to be fixed any time soon?

In case it helps: Here's a [repro gist](https://gist.github.com/7b227e44672555bb57662df5a5f367c4) and a screenshot of the resulting UX in Windows 10:

<img width="930" alt="image" src="https://user-images.githubusercontent.com/69829015/187833785-7eb05c98-b380-4e39-aab6-3a491a608fea.png">

**Comment by rzhao271** [comment]
I just discovered that we might be missing a `SetBackground()` call:
https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/ui/views/frame/glass_browser_caption_button_container.cc;l=172;bpv=1;bpt=1

Though I don't yet know how to make it so that the titlebar colour gets passed in.
The Electron equivalent of the code above is in https://github.com/electron/electron/blob/main/shell/browser/ui/views/win_caption_button_container.cc#L130.

**Comment by martijnhartlief** [comment]
I would like to add that:

- Changing the height property also sometimes show a the line below the buttons.
- Toggling properties with the window.setTitleBarOverlay function also changes whether a line shows below the buttons.
- Changing the symbolColor and color to rgba(0,0,0,0) will cause hovering to break. The hover color stays usually, not always.
- Losing focus of the window changes the appearance and will sometimes completely hide the buttons

Is there is no way to toggle WCO (like you can with MacOS)?
